syntax = "proto3";

import "google/protobuf/timestamp.proto";

package social_engine_grpc;

message Post {
    int64 id = 1;
    string title = 2;
    string text = 3;
    int64 user_id = 4;
    google.protobuf.Timestamp creation_time = 5;
}

message PostRequest {
    Post post = 1;
    int64 request_user_id = 2;
}

message PostResponse {
    Post post = 1;
    int64 code = 2;
    string description = 3;
}

service SocialEngine {
    rpc CreatePost(PostRequest) returns (PostResponse);
    rpc UpdatePost(PostRequest) returns (PostResponse);
    rpc DeletePost(PostRequest) returns (PostResponse);
    rpc GetPost(PostRequest) returns (PostResponse);
    rpc GetUserPosts(PostRequest) returns (stream PostResponse);
}
